VPC-Section-Formatting (#141)
C2Devel/terraform-provider-croccloud: docs/data-sources/internet_gateway.md

Reviewer: (optional) я бы предложил придерживаться того же принципа, что и пользовательской доке -- одно предложение, одна строка -- при разбиении текста

> ## Example Usage

Reviewer: Решили вроде писать как в предложении, много осталось в таком стиле, надо просмотреть и поменять

```suggestion
## Example usage
```

}
```

## Argument reference

The arguments of this data source act as filters for querying the available internet gateway.
The given filters must exactly match one internet gateway whose data will be exported as attributes.

> * `filter` - (Optional, Block) One or more name/value pairs to use as filters.

Reviewer: @PixarV предлагала добавить сам блок

}
```

## Argument reference

The arguments of this data source act as filters for querying the available internet gateway.

> The given filters must exactly match one internet gateway whose data will be exported as attributes.

Reviewer 2: Похожие блоки то появляются в датасорсах, то исчезают. Предлагаю заменить на стандартную формулировку `The following arguments are supported:`, а информацию о том, что отфильтровываться должен строго один ресурс, вынести в инфо или варнинг блок и продублировать в остальные датасорсы, которые отфильтровывают 1 ресурс

* `filter` - (Optional, [Block](#filter)) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-igws]
* `internet_gateway_id` - (Optional, String) The ID of the internet gateway.
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.

### filter

* `name` - (Required, String) The name of the filter. Names are case-sensitive.

> * `values` - (Required, List of strings) One or more filter values. The values are case-sensitive.

Reviewer 2: Names без артикля, The values с артиклем. Как будто должно быть одинаково

И ниже

Author: Оформил иначе @PixarV

## Attribute reference

All arguments except for the `filter` block are also exported as result attributes.

> If any fields are missing from the configuration, then this data source will populate them with data for the selected internet gateway.

Reviewer 2: Можно заменить на более распространенную формулировку `In addition to all arguments above, the following attributes are exported:`

Уточнение про `filter` можно убрать.

* `arn` - (String) The Amazon Resource Name (ARN) of the internet gateway.

> * `attachments` - (List of strings, [Block](#attachments)) List of VPC attachments to the internet gateway. The list can contain 0 or 1 element.

Reviewer 2: ```suggestion
* `attachments` - ([Block](#attachments)) List of VPC attachments to the internet gateway. The list can contain 0 or 1 element.
```

* `id` - (String) The ID of the internet gateway.

> * `owner_id` - (String) The ID of the project that the internet gateway belongs to.

Reviewer 2: ```suggestion
* `owner_id` - (String) The ID of the project that owns the internet gateway.
```

}
```

## Argument reference

The arguments of this data source act as filters for querying the available internet gateway.

~> **Note** The given filters must exactly match the resource whose data will be exported as attributes.

* `filter` - (Optional, [Block](#filter)) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-igws]
* `internet_gateway_id` - (Optional, String) The ID of the internet gateway.
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.

### filter

* `name` - (Required, String) The name of the filter.
    * _Constraints:_ Filter names are case-sensitive
* `values` - (Required, List of strings) One or more filter values.
    * _Constraints:_ Filter values are case-sensitive

## Attribute reference

If any fields are missing from the configuration, then this data source will populate them with data for the selected internet gateway.

In addition to all arguments above, the following attributes are exported:

* `arn` - (String) The Amazon Resource Name (ARN) of the internet gateway.

> * `attachments` - ([Block](#attachments)) List of VPC attachments to the internet gateway. The list can contain 0 or 1 element.

Reviewer 2: ```suggestion
* `attachments` - ([Block](#attachments)) The list of VPC attachments to the internet gateway. The list can contain 0 or 1 element.
```

> ## Argument Reference

Reviewer: здесь в т.ч.
```suggestion
## Argument reference
```

The arguments of this data source act as filters for querying the available internet gateway.
The given filters must match exactly one internet gateway whose data will be exported as attributes.

> * `filter` - (Optional, List of strings)) One or more name/value pairs to use as filters.

Reviewer: лишняя скобка
```suggestion
* `filter` - (Optional, List of strings) One or more name/value pairs to use as filters.
```

    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-igws]
* `internet_gateway_id` - (Optional, String) The ID of the internet gateway.
* `tags` - (Optional, Map of strings) Key-value pairs. Must exactly match pairs on the required resource.

## Attribute Reference

All arguments except `filter` block are also exported as result attributes.

> If any fields are missing from the configuration,

Reviewer: если выше делал предложение в строку, то надо было везде поменять

Author: Одно предложение - одна строка. Всюду пройдусь и поправлю.

then this data source will populate them with data for the selected internet gateway.

* `arn` - (String) The Amazon Resource Name (ARN) of the internet gateway.

> * `attachments` - (List of strings) List of VPC attachments to the internet gateway. It can contain 0 or 1 element. 

Reviewer: Это не List of strings. Как Рита предлагает:
```suggestion
* `attachments` - (List of [attachments](#attachments)) VPC attachments to the internet gateway. The list can contain 0 or 1 element. 
```
 Если решим делать так, то для всех блоковых атрибутов/аргументов надо поменять аналлгично

>   See [attachments](#attachments) below for more details.

Reviewer: это удалить тогда, если будет исправлено, как предлагается выше

* `filter` - (Optional) One or more name/value pairs to use as filters.
    * _Valid values:_ See supported names and values in [EC2 API documentation][describe-igws]

> * `internet_gateway_id` - (Optional) (String) The ID of the internet gateway.

Reviewer: Если добавлять типы данных, то тогда везде придется добавлять, в частности в пиаре по EBS.
Чтобы уменьшит число скобок, предлагаю как у Яндекса, все объединить в одни скобки через зпт
```suggestion
* `internet_gateway_id` - (Optional, String) The ID of the internet gateway.
```

Author: Тогда в одних скобках будет дофига всего, и Editable или Forces new resource, и тип данных, как-то перегружено
@DiGa66

Author: Сделал всё в одних скобках

> * `tags` - (Optional) (Map of tags) Key-value map of tags. Each tag must exactly match a tag on the desired internet gateway.

Reviewer: практически повтор текстовый, к тому же, насколько я понимаю, map уже подразумевает набор ключ:значение
И есть такой тип Map of keys? Не Map of strings?
    
    
```suggestion
* `tags` - (Optional) (Map of strings) Each tag must exactly match a tag on the desired internet gateway.
```

Author: Всюду про tags в секции vpc сделал единообразно, как обсуждали

> * `arn` - The Amazon Resource Name (ARN) of the internet gateway.

Reviewer: в нашей доке у ARN тип string

> * `attachments` - List of VPC attachments to the internet gateway. It can contain 0 or 1 element.

Reviewer: filter, arn, attachment -- здесь сталкиваемся со сложностями последовательного указания типа данных, так стоило ли начинать?
у Яндекса подобная структура как для attachments указывается как [Block] в квадратных скобках. И ее можно сделать ссылкой, это позволит убрать фразу со ссылкой в конце описания
@PixarV  что думаешь по поводу указания типов в целом

и надо договориться, мы будем оставлять цифры или писать прописью  в аналогичных ситуациях -- zero or one element

Reviewer 2: Если решили вводить типы данных, то их нужно указывать для каждого аргумента (а по хорошему еще и для каждого атрибута).

<block> можно считать типом данных, т.е. тут было бы List of `attachments` (attachments - имя параметра в конфигурации). Думаю, что будет здорово сделать attachments ссылкой

Author: Для `filter` прописал List of strings. Для `attachments` в дескрипшене есть ссылка на блок с дополнительными аргументами. ИМХО в скобках со свойствами будет странно делать ссылку на этот блок, мы нигде так не делаем.
@DiGa66

Reviewer: мы и типы данных нигде не добавляли раньше, можно договориться делать ссылкой, тогда предложение про блок не понадобится
